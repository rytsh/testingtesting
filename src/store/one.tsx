import { createStore, applyMiddleware, compose } from "redux";
import ReduxThunk from 'redux-thunk'

const initialState = {
  calculator: {
    screen: "0",
    history: [],
    dec: false,
    changen: true
  },
  posts : {
    text: "",
    content: [],
    isLoading: false,
    error: null
  },
  info: ""
}

function reducer(state:any = initialState, action: any): object {
  switch (action.app) {
    case "CALCULATOR":
      switch (action.type) {
        case "HOLD":
          return {
            ...state,
            calculator: {
              ...state.calculator,
              screen: action.value,
              dec: action.dec,
              changen: action.changen
            }
          };
        case "EQUAL":
          if (state.calculator.history.length === 0 || state.calculator.history[state.calculator.history.length-1].calc !== action.history)
            state.calculator.history.push({calc: action.history});

          return {
            ...state,
            calculator: {
              ...state.calculator,
              dec: action.dec,
              changen: action.changen,
              screen: action.value,
              history: [...state.calculator.history]
            }
          };
        case "DEL":
          return {
            ...state,
            calculator: {
              ...state.calculator,
              history: state.calculator.history.filter((_:any, index:number) => {return index !== action.index})
            }
          };
        case "DEL-LAST":
          state.calculator.history.pop();
          return {
            ...state,
            calculator: {
              ...state.calculator,
              history: [...state.calculator.history]
            }
          };
        default:
          return state;
      }
    case "POSTS":
      switch (action.type) {
        case "TEXT":
          return {
            ...state,
            posts : {
              ...state.posts,
              text: action.text,
            }
          }
        case "GET_POSTS_BEGIN":
          return {
            ...state,
            posts : {
              ...state.posts,
              isLoading: true,
              error: null
            }
          }
        case "GET_POSTS_SUCCESS":
          return {
            ...state,
            posts : {
              ...state.posts,
              content: action.content,
              isLoading: false,
            }
          }
        case "GET_POSTS_ERROR":
          return {
            ...state,
            posts : {
              ...state.posts,
              error: action.error,
              isLoading: false,
            }
          }
        default:
          return state
      }
      case "INFO":
        switch (action.type) {
          case "READ":
            return {
              ...state,
              info : action.data
            }
          default:
            return state
        }
    default:
      return state
  }
  
}

declare global {
  interface Window { __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: any; }
}

const composeEnhancers =
  typeof window === 'object' &&
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?   
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
      // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
    }) : compose;

const enhancer = composeEnhancers(
  applyMiddleware(ReduxThunk),
  // other store enhancers if any
);

const store = createStore(reducer, enhancer);

export default store
