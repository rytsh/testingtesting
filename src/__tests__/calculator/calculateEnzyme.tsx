import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import store from "store/one";

import Calculator from "components/calculator/View";

Enzyme.configure({ adapter: new Adapter});

function mWithStore(
  ui:any,
  storex:any = store
) {
  return {
    wrap: Enzyme.mount(<Provider store={storex}>{ui}</Provider>) as Enzyme.ReactWrapper,
    storex,
  }
}

function rWithStore(
  ui:any,
  storex:any = store
) {
  return {
    wrap: Enzyme.render(<Provider store={storex}>{ui}</Provider>) as Cheerio,
    storex,
  }
}


describe('Enzyme Calculator', () => {
  it('should render app', ()=> {
    const wrapper = Enzyme.shallow(<Calculator/>);
    // console.log(wrapper.debug());
    expect(wrapper.find('.title').exists()).toBe(true);
    expect(wrapper.find('p[data-testid="title"]').text()).toBe("Calculator with history");
    expect(wrapper.find('[data-testid="title"]').text()).toBe("Calculator with history");
    // enzyme-to-json toJson
    expect(wrapper).toMatchSnapshot();;

    // get probs values
    // wrapper.instance().props.address
    // set probs, probs should be valid
    // wrapper.setProps({hide: true});
    // expect(wrapper.props().hide).toBeTruthy();
  })

  // it('calls component did mount', ()=> {
    // jest.spyOn(Calculator.prototype, 'componentDidMount');
    // const wrapper:any = Enzyme.shallow(<Calculator/>);
    // expect(wrapper.prototype.componentDidMount.mock.calls.length).toBe(1);
  // })

  it('mount', ()=>{
    const wrapper = mWithStore(<Calculator/>);
    // console.log(wrapper.wrap.debug());
    wrapper.wrap.find('[value="9"]').simulate("click");
    // wrapper.storex
    // wrapper.wrap.update();
    // console.log(wrapper.wrap.debug());
  })

  // it('render', ()=>{
    // this is just a DOM check
    // const wrapper = rWithStore(<Calculator/>);
    // console.log(wrapper.wrap.debug());
    // wrapper.wrap.find('[value="9"]').simulate("click");
    // console.log(wrapper.wrap.debug());
  // })
})
