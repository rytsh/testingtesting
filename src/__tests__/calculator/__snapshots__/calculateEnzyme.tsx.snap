// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enzyme Calculator should render app 1`] = `ShallowWrapper {}`;
